api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -51,14 +50,13 @@ message UpstreamTlsContext {
   // interacts with other validation options.
   bool auto_host_sni = 6;
 
-  // If true, replace any Subject Alternative Name validations with a validation for a DNS SAN matching
-  // the SNI value sent. Note that the validation will be against the actual requested SNI, regardless of how it
-  // is configured.
+  // If true, replaces any Subject Alternative Name (SAN) validations with a validation for a DNS SAN matching
+  // the SNI value sent. The validation uses the actual requested SNI, regardless of its configuration.
   //
-  // For the common case where an SNI value is sent and it is expected that the server certificate contains a SAN
-  // matching that SNI value, this option will do the correct SAN validation.
+  // This option ensures correct SAN validation when an SNI value is provided and expected to match the server
```

Comment:
I think it should somehow note that this is the common/normal case

Reply:
Changed it to, _"For common cases where an SNI value is present and the server certificate should include a corresponding SAN, this option ensures the SAN is properly validated."_